apps/test/unit/templates/studioHomepages/RecentCoursesTest.js

```diff
@@ -7,149 +7,210 @@ import SetUpCourses from '@cdo/apps/templates/studioHomepages/SetUpCourses';
+              type={Notification.NotificationType.course}
+              notice="Find a course"
+              details="Try new courses to add them to your homepage."
+              buttonText="Find a course"
+              buttonLink="/courses"
+              dismissible={false}
               isRtl={false}
-              isTeacher
             />
-          </ContentContainer>
-        </div>
-      );
-    });
+          </div>
+        </ContentContainer>
+      </div>
+    );
+    expect(wrapper.find('SetUpCourses').exists()).to.be.false;
+    expect(wrapper.find('CourseCard').exists()).to.be.false;
+    expect(wrapper.find('SeeMoreCourses').exists()).to.be.false;
+  });
 
-    it('shows only cards for each course when there are 4 or less courses', () => {
-      const wrapper = shallow(
-        <RecentCourses
-          courses={courses}
-          isTeacher
+  it('shows TopCourse and 2 CourseCards when there are 3 courses', () => {
+    const wrapper = shallow(
+      <RecentCourses
+        courses={courses}
+        topCourse={topCourse}
+        isTeacher
+        isRtl={false}
+      />
+    );
+    expect(wrapper).to.containMatchingElement(
+      <div>
+        <ContentContainer
           heading="My Courses"
           isRtl={false}
-        />
-      );
-      expect(wrapper).to.containMatchingElement(
-        <div>
-          <ContentContainer
-            heading="My Courses"
+        >
+          <TopCourse
             isRtl={false}
-          >
-            <div key={0}>
-              <CourseCard
-                title={courses[0].title}
-                description={courses[0].description}
-                link={courses[0].link}
-                isRtl={false}
-              />
-              <div>.</div>
-            </div>
-            <div key={1}>
-              <CourseCard
-                title={courses[1].title}
-                description={courses[1].description}
-                link={courses[1].link}
-                isRtl={false}
-              />
-            </div>
-            <div>
-              <Notification
-                type={Notification.NotificationType.course}
-                notice="Find a course"
-                details="Try new courses to add them to your homepage."
-                buttonText="Find a course"
-                buttonLink="/courses"
-                dismissible={false}
-                isRtl={false}
-              />
-            </div>
-          </ContentContainer>
-        </div>
-      );
-    });
+            assignableName={topCourse.assignableName}
+            lessonName={topCourse.lessonName}
+            linkToOverview={topCourse.linkToOverview}
+            linkToLesson={topCourse.linkToLesson}
+          />
+          <div key={0}>
+            <CourseCard
+              title={courses[0].title}
+              description={courses[0].description}
+              link={courses[0].link}
+              isRtl={false}
+            />
+            <div>.</div>
```

Comment:
Given how explicit this declaration of the DOM is, down to a period, I wonder if this form of matching is a little brittle?